azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualMachineScaleSetApiMockTest.java

```diff
@@ -0,0 +1,257 @@
+      //Create properties object
+      final NetworkInterfaceCardProperties networkInterfaceCardProperties = NetworkInterfaceCardProperties
+         .builder()
+         .ipConfigurations(
+            Arrays.asList(IpConfiguration.create(ipConfigurationName, null, null, null, IpConfigurationProperties
+               .create(null, null, "Dynamic", subnetId, null, null, null)))).build();
+
+      final Map<String, String> tags = ImmutableMap.of("jclouds", "livetest");
+      return api.getNetworkInterfaceCardApi(resourceGroupName).createOrUpdate(networkInterfaceCardName, locationName, networkInterfaceCardProperties, tags);
+   }
+
+
+   private ExtensionProfile getExtensionProfile() {
+      List<Extension> extensions = new ArrayList<Extension>();
+
+      List<String> uris = new ArrayList<String>();
+      uris.add("https://mystorage1.blob.core.windows.net/winvmextekfacnt/SampleCmd_1.cmd");
+      ExtensionProfileSettings extensionProfileSettings = ExtensionProfileSettings.create(uris, "SampleCmd_1.cmd");
+
+      Map<String, String> protectedSettings = new HashMap<String, String>();
+      protectedSettings.put("StorageAccountKey", "jclouds-accountkey");
+
+      ExtensionProperties extensionProperties = ExtensionProperties.create("Microsoft.compute", "CustomScriptExtension",
+         "1.1", false, extensionProfileSettings,
+         protectedSettings);
+
+      Extension extension = Extension.create("extensionName", extensionProperties);
+      extensions.add(extension);
+
+      return ExtensionProfile.create(extensions);
+   }
+
+   private VirtualMachineScaleSetVirtualMachineProfile getVirtualMachineProfile() {
+      return VirtualMachineScaleSetVirtualMachineProfile.create(getLinuxStorageProfile_Default(), getOSProfile(), getNetworkProfile(), getExtensionProfile());
+   }
+
+   private VirtualMachineScaleSetPlan getVirtualMachineScaleSetPlan() {
+      return VirtualMachineScaleSetPlan.create("thinkboxsoftware", "deadline-slave-7-2", "deadline7-2");
+   }
+
+   private String getVMSSName() {
+      return String.format("%3.24s", System.getProperty("user.name") + "mock" + this.getClass().getSimpleName()).toLowerCase().substring(0, 15);
+   }
+
+   private String getSubnet() {
+      return String.format("s-%s-%s", this.getClass().getSimpleName().toLowerCase(), System.getProperty("user.name"));
+   }
+
+   private Subnet createDefaultSubnet()  {
+      server.enqueue(jsonResponse("/getonesubnet.json").setResponseCode(200));
+      final SubnetApi subnetApi = api.getSubnetApi("azurearmtesting", "myvirtualnetwork");
+      Subnet subnet = subnetApi.get("mysubnet");
+      String path = String.format("/subscriptions/%s/resourcegroups/%s/providers/Microsoft.Network/virtualNetworks/%s/subnets/%s?%s", subscriptionid, resourcegroup, virtualNetwork, subnetName, apiVersion);
+      try {
+         assertSent(server, "GET", path);
+      } catch (InterruptedException e) {
+         e.printStackTrace();
+      }
+      return subnet;
+   }
+
+
+}
```

Comment:
Tests still missing here.

Reply:
Tests added...

Reply:
Mock tests are intended to verify that the API methods produce the expected requests. We test the API against a mock webserver that records the requests it receives, and then we assert that the expected ones were generated.
The tests in this class must exercise all the methods in the API (list, createOrUpdate, get, delete), and then validate that all produced requests are the expected ones.
You can use the [DiskApiMockTest](https://github.com/jclouds/jclouds-labs/blob/master/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/DiskApiMockTest.java) as a model.